License2Deploy/rolling_deploy.py

```diff
@@ -0,0 +1,60 @@
+import boto.ec2.autoscale as asg
+import re
+
+
+
+def return_autoscalinggroup_desired_size(auto_scaling_group):
+    """ AWS Call: return the  minsize of the AutoScaling Group of the Microservice"""
+    return auto_scaling_group.min_size
+
+def calculate_new_desired_ec2_count(current_desired_ec2_count):
+    """ By default, we double the instance Count """
+    return 2 * current_desired_ec2_count
+
+def update_desired_ec2_count_in_autoscalinggroup(auto_scaling_group, ec2_count):
+    auto_scaling_group.set_capacity(ec2_count)
+    """ AWS Call: update the desired_size of the AutoScaling Group of the Microserice"""
+    pass
+
+def healthcheck_new_instances():
+    """ AWS Call: Check whether the new instances are working good """
+    return True
+
+
+def update_ami_id_with_last_successfull():
```

Comment:
I don't think this method is necessary. If the healtchecks fail, the job should return an exit code of something other than zero which will in turn trigger jenkins to run the job again with the last known ami id. The ami id should be passed in as a parameter to this script using sys.argv, argparse, etc.